server/tests/e2e.rs

```diff
@@ -46,16 +52,86 @@ async fn start_deqs_server(
     initial_scis: &[SignedContingentInput],
     logger: &Logger,
 ) -> (TestServer, TestQuoteBook, DeqsClientApiClient) {
+    let (msg_bus_tx, msg_bus_rx) = broadcast::channel::<Msg>(MSG_BUS_QUEUE_SIZE);
+    let remove_quote_callback = get_callback_function(msg_bus_tx.clone());
     let internal_quote_book = InMemoryQuoteBook::default();
-    let synchronized_quote_book =
-        SynchronizedQuoteBook::new(internal_quote_book, ledger_db.clone());
+    let synchronized_quote_book = SynchronizedQuoteBook::new(
+        internal_quote_book,
+        ledger_db.clone(),
+        remove_quote_callback,
+        logger.clone(),
+    );
+
+    let (deqs_server, client_api) = start_deqs_server_for_quotebook(
+        initial_scis,
+        &synchronized_quote_book,
+        p2p_bootstrap_from,
+        msg_bus_tx,
+        msg_bus_rx,
+        logger,
+    )
+    .await;
+
+    (deqs_server, synchronized_quote_book, client_api)
+}
 
+async fn start_in_memory_deqs_server(
+    p2p_bootstrap_from: &[&TestServer],
+    initial_scis: &[SignedContingentInput],
+    logger: &Logger,
+) -> (
+    Server<InMemoryQuoteBook>,
+    InMemoryQuoteBook,
+    DeqsClientApiClient,
+) {
+    let (msg_bus_tx, msg_bus_rx) = broadcast::channel::<Msg>(MSG_BUS_QUEUE_SIZE);
+    let internal_quote_book = InMemoryQuoteBook::default();
+
+    let (deqs_server, client_api) = start_deqs_server_for_quotebook(
+        initial_scis,
+        &internal_quote_book,
+        p2p_bootstrap_from,
+        msg_bus_tx,
+        msg_bus_rx,
+        logger,
+    )
+    .await;
+
+    (deqs_server, internal_quote_book, client_api)
+}
+
+fn get_callback_function(
```

Comment:
Maybe this can become a static method inside `Server`, and then it can be reused by the tests code and the server's `main()`?